homeassistant/components/bme280spi/sensor.py

```diff
@@ -0,0 +1,169 @@
+"""Support for BME280 temperature, humidity and pressure sensor."""
+from datetime import timedelta
+from functools import partial
+import logging
+
+from bme280spi import BME280  # pylint: disable=import-error
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS,
+    CONF_NAME,
+    PERCENTAGE,
+    TEMP_FAHRENHEIT,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.util.temperature import celsius_to_fahrenheit
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SPI_DEV = "spi_dev"
+CONF_SPI_BUS = "spi_bus"
+CONF_OVERSAMPLING_TEMP = "oversampling_temperature"
+CONF_OVERSAMPLING_PRES = "oversampling_pressure"
+CONF_OVERSAMPLING_HUM = "oversampling_humidity"
+CONF_T_STANDBY = "time_standby"
+CONF_FILTER_MODE = "filter_mode"
+
+DEFAULT_NAME = "BME280 Sensor"
+DEFAULT_SPI_DEV = 0
+DEFAULT_SPI_BUS = 0
+DEFAULT_OVERSAMPLING_TEMP = 1  # Temperature oversampling x 1
+DEFAULT_OVERSAMPLING_PRES = 1  # Pressure oversampling x 1
+DEFAULT_OVERSAMPLING_HUM = 1  # Humidity oversampling x 1
+DEFAULT_T_STANDBY = 5  # Tstandby 5ms
+DEFAULT_FILTER_MODE = 0  # Filter off
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=3)
+
+SENSOR_TEMP = "temperature"
+SENSOR_HUMID = "humidity"
+SENSOR_PRESS = "pressure"
+SENSOR_TYPES = {
+    SENSOR_TEMP: ["Temperature", None],
+    SENSOR_HUMID: ["Humidity", PERCENTAGE],
+    SENSOR_PRESS: ["Pressure", "mb"],
+}
+DEFAULT_MONITORED = [SENSOR_TEMP, SENSOR_HUMID, SENSOR_PRESS]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer allow new integrations to use a platform config section or old integrations to do changes to the platform config section. See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md.

Please refactor this integration to have the config schema in the base integration module and the config yaml section under the top level integration domain key.

Reply:
Also, this should not be a new integration but instead a config option for the existing bme280 integration.

Reply:
The best option would be close this PR and refactor bme280 integration. Then add the config option with SPI. Can I use some example for refactor integration to have the config schema in the base integration module?

Reply:
Here's an example of the config schema for an integration that configures platforms from the top level config key of the integration:
https://github.com/home-assistant/core/blob/c1e788e6655284ef4276638665948d9602bac63f/homeassistant/components/netgear_lte/__init__.py#L83-L105

Here's how to load a platform:
https://github.com/home-assistant/core/blob/c1e788e6655284ef4276638665948d9602bac63f/homeassistant/components/netgear_lte/__init__.py#L244-L248